### PR TITLE
docs/nia: update task api

### PR DESCRIPTION
**Pull request by lornasong**
Commits:
- Convert existing API page into an 'Overview' page and reorganize so that we can have a API directory with a page for different APIs
- Move existing Status API information from 'Overview' page to its own 'Status' page
- Add a 'Tasks' API page with details on update task API

Preview
 - [Newly moved overview page](https://consul-beyvkyloi-hashicorp.vercel.app/docs/nia/api/api-overview)
 - [Newly moved status page](https://consul-beyvkyloi-hashicorp.vercel.app/docs/nia/api/status)
 - [New task subcommand page](https://consul-beyvkyloi-hashicorp.vercel.app/docs/nia/api/tasks)

### CODE REVIEW

**Review by eikenb — APPROVED** [review]
Makes sense to me.

File: website/content/docs/nia/api/api-overview.mdx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Example: `localhost:8558/v1/status`
+
+### Error
+
+Successful API requests will receive a 2XX success status code. For other unsuccessful status codes, when possible, more details will be provided in a response body. The response will be a JSON map with an "error" key.
+
+Example: Status 400 Bad Request
+```json
+{
+  "error": "example error message: unsupported status parameter value"
```
Comment (findkim) [inline website/content/docs/nia/api/api-overview.mdx:30]:
```suggestion
  "error": {
    "message": "example error message: unsupported status parameter value"
  }
```

Comment (findkim) [reply website/content/docs/nia/api/api-overview.mdx:30]: If you don't mind adding a change for me too! I forgot until I saw the diffs

**Review by findkim — APPROVED** [review]

File: website/content/docs/nia/api/api-overview.mdx (hunk 1 above)
Comment (lornasong) [reply website/content/docs/nia/api/api-overview.mdx:30]: Thanks for thinking of this! Of course I don't mind. I actually have another PR prepared of miscellaneous API changes and have a commit for this. If it's ok with you, I'll leave this change for that PR so I can avoid waiting and rechecking vercel?

Comment (findkim) [reply website/content/docs/nia/api/api-overview.mdx:30]: That works too thank you

**Pull request merged**